libraries/AP_BattMonitor/AP_BattMonitor_DJIBattery.cpp

```diff
@@ -0,0 +1,105 @@
+#include <AP_HAL/AP_HAL.h>
+#include "AP_BattMonitor_DJIBattery.h"
+#include <GCS_MAVLink/GCS.h>
+#include <AP_SerialManager/AP_SerialManager.h>
+#include <AP_Math/crc.h>
+
+/*
+  "battery" monitor for DJI Battery via serial port.
+ */
+extern const AP_HAL::HAL& hal;
+
+/// Constructor
+AP_BattMonitor_DJIBattery::AP_BattMonitor_DJIBattery(AP_BattMonitor &mon, AP_BattMonitor::BattMonitor_State &mon_state, AP_BattMonitor_Params &params) :
+    AP_BattMonitor_Backend(mon, mon_state, params)
+{
+    const AP_SerialManager &serial_manager = AP::serialmanager();
+    
+    if ((port = serial_manager.find_serial(AP_SerialManager::SerialProtocol_DJIBattery, 0))) {
+        last_send_ms = AP_HAL::micros();
+        delay_time_ms = 1000000;
+        Count = 0;
+        _PortAvailable = true;
+        gcs().send_text(MAV_SEVERITY_INFO,"DJI Battery Port Attached");
+    } else { 
+    _PortAvailable = true;
```

Comment:
no biggies but a small formatting issue here.. should be indented a bit more I think